dsl/src/test/scala/org/http4s/dsl/PathInHttpRoutesSuite.scala

```diff
@@ -71,8 +72,8 @@ class PathInHttpRoutesSuite extends Http4sSuite {
       Ok(s"list: ${l.mkString(",")}, start: $s, limit: ${m.l}, term: $t, decimal=$d")
     case GET -> Root / "app" :? OptCounter(c) =>
       Ok(s"counter: $c")
-    case GET -> Root / "required" :? RequiredCounter(withCount) =>
-      withCount(vc => Ok(s"counter: $vc"))
+    case GET -> Root / "required" :? RequiredCounter(c) :? RequiredLimit(l) =>
+      (c, l).tupled.respondWith { case (vc, vl) => Ok(s"counter: $vc, limit: $vl") }
```

Comment:
I'm way of ad hoc syntactic extensions, but trying to think of an alternative.  I like that you solved the composition problem.

Thinking out loud, and not compiling:

```scala
respondValid((c, l).mapN { case (vc, vl) => Ok(s"counter: $vc, limit: $vl") })
```